app/services/locations/updater.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Locations
+  class Updater
+    YOI_NOMIS_AGENCIES = %w[WYI WNI CKI PRI FMI].freeze
```

Comment:
lets move the actual list into a yaml file, rather than hard-coded

Reply:
I'm not convinced by this - it would still be hardcoded but in yaml form, and mean slightly more complex code to achieve the same thing. We also have other existing data services that set up data without using a yaml file. Either way it means a PR and a deployment to invoke a change to the data.

Reply:
moving it to a yaml (e.g. `YAML.safe_load(File.read('./lib/tasks/data/young_offender_institutions.yml'))`) would help keep all our reference data in one place, rather than having to scour the code for the list when the inevitable changes come through. I don't think we have other reference data which is hard coded?

Reply:
There are quite a few actually... see:
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/allocation_complex_cases/importer.rb 
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/personal_care_needs/importer.rb
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/nomis_alerts/importer.rb
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/identifier_types/importer.rb
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/genders/importer.rb
https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/assessment_questions/importer.rb

I believe it's important to keep things as simple as possible, so really don't agree with moving this array of 5 things to a YAML file. If over time the needs of the service become more complex and we need additional data this can be revised of course but right now this is the simplest thing that works (tm) 😁

Reply:
ok, fair enough 👍